### PR TITLE
Beacon api: propoerly submit blind block

### LINKED ISSUE #11477: POST /eth/v1/beacon/blinded_blocks Does Not Work
# 🐞 Bug Report

### Description

The `POST /eth/v1/beacon/blinded_blocks` endpoint in Prysm does not work correctly. It accepts a blinded beacon block and then passes it through the normal `submitBlock` function that are used for all other block types. This function broadcasts over p2p, processes in the blockchain service, and more. However, a blinded block _must_ be converted to full format before it can be passed through `submitBlock`. Otherwise, the processing will fail. Moreover, Prysm will try to broadcast the blinded block over p2p which is going to return an error.

### Has this worked before in a previous version?

No

## 🔬 Minimal Reproduction

Submit a blinded beacon block to Prysm via the standard API

### PR DESCRIPTION
**Pull request by terencechain**
**What type of PR is this?**

> Bug fix

**What does this PR do? Why is it needed?**

When using beacon API to submit a blind block, it must first go through mev boost/relayer/builder to retrieve the full block. This PR made sure of that. The status duo is to broadcast the blind block, and that's not going to work

**Which issues(s) does this PR fix?**

Fixes #11477

**Other notes for review**

N/A

### CODE REVIEW

**Review by rkapka — CHANGES_REQUESTED** [review]
We also have to fix `SubmitBlindedBlockSSZ`

**Comment by terencechain** [comment]
> We also have to fix `SubmitBlindedBlockSSZ`

Fixed. thanks!

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 1]
```diff
@@ -342,6 +343,19 @@ func (bs *Server) SubmitBlindedBlockSSZ(ctx context.Context, req *ethpbv2.SSZCon
 	if err != nil {
 		return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not unmarshal request data into block: %v", err)
 	}
+
+	if block.IsBlinded() {
+		b, err := block.PbBlindedBellatrixBlock()
+		if err != nil {
+			return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not get blind block: %v", err)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks.go:350]:
```suggestion
			return &emptypb.Empty{}, status.Errorf(codes.Internal, "Could not get blinded block: %v", err)
```

File: beacon-chain/rpc/eth/beacon/blocks.go [hunk 2]
```diff
@@ -875,24 +889,22 @@ func (bs *Server) submitBellatrixBlock(ctx context.Context, bellatrixBlk *ethpbv
...
-
-	root, err := blindedBellatrixBlk.HashTreeRoot()
+	_, err = bs.V1Alpha1ValidatorServer.ProposeBeaconBlock(ctx, &eth.GenericSignedBeaconBlock{
+		Block: &eth.GenericSignedBeaconBlock_BlindedBellatrix{
+			BlindedBellatrix: b,
+		},
+	})
 	if err != nil {
-		return status.Errorf(codes.InvalidArgument, "Could not tree hash block: %v", err)
+		return status.Errorf(codes.Internal, "Could not propose blind block: %v", err)
```
Comment (rkapka) [inline beacon-chain/rpc/eth/beacon/blocks.go:905]:
```suggestion
		return status.Errorf(codes.Internal, "Could not propose blinded block: %v", err)
```

**Review by rkapka — APPROVED** [review]

**Pull request merged**